modules/fun.js

```diff
@@ -0,0 +1,450 @@
+      }
+      board[x].push(getMineCount(x, y));
+    }
+  }
+  const output = board.map(row => row.map(num => `||${num == 9 ? mineSweeperEmojis[`bomb`] : mineSweeperEmojis[num]}||`).join(``)).join(`\n`);
+  field = (`**Mines: ${mineCount}** (Tip: Corners are never mines)\n${output}`);
+  // we need to split it up because only 99 emoji per message limit for some reason.
+  let degradingField = field;
+  function countEmoji(text) {
+    const emojiRegex = new RegExp(`(${Object.values(mineSweeperEmojis).map(emoji => emoji.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')).join(`|`)})`, 'g');
+    const emoji = text.match(emojiRegex);
+    return emoji?.length || 0;
+  }
+  if (countEmoji(field) <= 99) {
+    return int.editReply(field); // No splitting needed
+  }
+  while (countEmoji(degradingField) > 99) {
+    let segment = ``;
+    while (countEmoji(segment + degradingField.substring(0, degradingField.indexOf(`\n`) >= 0 ? degradingField.indexOf(`\n`) : degradingField.length)) <= 99) {
+      segment += degradingField.substring(0, (degradingField.indexOf(`\n`) >= 0 ? degradingField.indexOf(`\n`) : degradingField.length) + 1);
+      degradingField = degradingField.substring((degradingField.indexOf(`\n`) >= 0 ? degradingField.indexOf(`\n`) : degradingField.length) + 1);
+    }
+    if (segment + degradingField == field) {
+      await int.editReply(segment);
+    } else {
+      await int.channel.send(segment);
+    }
+  }
+  return int.channel.send(degradingField);
+}
+/**
+ * function slashFunRoll
+ * @param {Discord.ChatInputCommandInteraction} int a /fun roll interaction
+ */
+async function slashFunRoll(int) {
+  let dice = int.options.getInteger('dice');
+  let sides = int.options.getInteger('sides');
+  let modifier = int.options.getInteger('modifier');
+  if (!dice) dice = 1;
+  if (!sides) sides = 6;
+  if (!modifier) modifier = 0;
+  /** @type {string[][]} */
+  const rolls = [];
+  let total = 0;
+  rolls[sides] = [];
+  const num = dice;
+  if (num && num <= 10000) {
+    for (let i = 0; i < num; i++) {
+      const val = Math.ceil(Math.random() * sides);
+      rolls[sides].push((i == 0 ? `**d${sides}:** ` : ``) + val);
+      total += val;
+    }
+  } else {
+    return int.editReply(`I'm not going to roll *that* many dice... 🙄`);
+  }
+  if (modifier) {
+    total += dice;
+    rolls[sides].push(`**${dice}**`);
+  }
+  if (rolls.length > 0) {
+    return int.editReply(`You rolled ${dice}d${sides} and got:${total}\n` +
+        ((rolls.reduce((a, c) => a + c.length, 0) > 20) ? `` : ` ( ${rolls.reduce((a, c) => a + c.join(`, `) + `; `, ``)})`));
+  } else {
+    return int.editReply(`you didn't give me anything to roll.`);
+  }
+}
+/**
+ * function ball8
+ * @param {Discord.ChatInputCommandInteraction} int a /fun 8ball interaction
+ */
+async function slashFun8ball(int) {
+  const question = int.options.getString(`question`);
+  if (!question || !question.endsWith(`?`)) {
+    return int.editReply(`you need to ask me a question, silly.`);
```

Comment:
This is uh... kinda silly.
```js
if (...) {
  return 
} else {
  ...
}
```

Reply:
I do not see what is silly about it. ?using else so that its more readable instead of relying on return?
?the requirement of "?" at the end? what about this code is silly?

Reply:
or do you mean how I put a return in there instead of just inverting the condition from the start and putting the return at the end? this is much more readable because the check and the fail are right next to eachother and the rest of the code is just one or two lines further down as opposed to having the check and fail actions seperated by 25 lines and hard to locate.

Reply:
first one, got it. I'll stand by my code being more readable, but if you wish for it to be like such then so-be-it.